Lib/shlex.py

```diff
@@ -319,6 +319,9 @@ def join(split_command):
 
 def quote(s):
     """Return a shell-escaped version of the string *s*."""
+    if not isinstance(s, str):
+        raise TypeError(f"expected str, not {type(s)!r}")
```

Comment:
```suggestion
    if not isinstance(s, str):
        raise TypeError(f"expected string object, got {type(s).__name__!r}")
```

So that the error message remains identical.